src/app/screens/Nostr/ConfirmDecrypt.tsx

```diff
@@ -1,52 +1,64 @@
-import ConfirmOrCancel from "@components/ConfirmOrCancel";
 import Container from "@components/Container";
 import PublisherCard from "@components/PublisherCard";
-import { PopiconsCheckLine } from "@popicons/react";
 import { useState } from "react";
 import { useTranslation } from "react-i18next";
+import ConfirmOrCancel from "~/app/components/ConfirmOrCancel";
+import ContentMessage from "~/app/components/ContentMessage";
+import PermissionModal from "~/app/components/Permissions/PermissionModal";
+import PermissionSelector from "~/app/components/Permissions/PermissionSelector";
 import ScreenHeader from "~/app/components/ScreenHeader";
-import Checkbox from "~/app/components/form/Checkbox";
+import toast from "~/app/components/Toast";
 import { useNavigationState } from "~/app/hooks/useNavigationState";
-import { USER_REJECTED_ERROR } from "~/common/constants";
 import msg from "~/common/lib/msg";
-import { OriginData } from "~/types";
+import { OriginData, PermissionOption } from "~/types";
 
 function NostrConfirmDecrypt() {
   const { t } = useTranslation("translation", {
     keyPrefix: "nostr",
   });
   const { t: tPermissions } = useTranslation("permissions");
   const { t: tCommon } = useTranslation("common");
+
   const navState = useNavigationState();
   const origin = navState.origin as OriginData;
 
   const [loading, setLoading] = useState(false);
 
-  const [rememberPermission, setRememberPermission] = useState(true);
-
-  function confirm() {
-    setLoading(true);
-    msg.reply({
-      confirm: true,
-      rememberPermission,
-      blocked: false,
-    });
-    setLoading(false);
-  }
+  const [modalOpen, setModalOpen] = useState(false);
+  const [permissionOption, setPermissionOption] = useState<PermissionOption>(
+    PermissionOption.ASK_EVERYTIME
+  );
 
-  function reject(event: React.MouseEvent<HTMLAnchorElement>) {
-    event.preventDefault();
-    msg.error(USER_REJECTED_ERROR);
+  async function confirm() {
+    try {
+      setLoading(true);
+      msg.reply({
+        confirm: true,
+        permissionOption: permissionOption,
+        blocked: false,
+      });
+    } catch (e) {
+      console.error(e);
+      if (e instanceof Error) toast.error(`${tCommon("error")}: ${e.message}`);
+    } finally {
+      setLoading(false);
+    }
   }
 
-  async function block(event: React.MouseEvent<HTMLAnchorElement>) {
-    event.preventDefault();
-    await msg.request("addBlocklist", {
-      domain: origin.domain,
-      host: origin.host,
-    });
-    alert(`Added ${origin.host} to the blocklist, please reload the website`);
-    msg.error(USER_REJECTED_ERROR);
+  function reject(e: React.MouseEvent<HTMLAnchorElement>) {
+    try {
+      setLoading(true);
+      msg.reply({
```

Comment:
Possibly rejections have to be answered with msg.error(), that's probably why a signature is still delivered although user clicks "deny".